smithy4s-fetch.scala

```diff
@@ -233,7 +233,14 @@ private[smithy4s_fetch] case class SimpleRestJsonCodecs(
       uriScheme,
       uri.host,
       uri.port.toIntOption,
-      uri.pathname.split("/"),
+      uri.pathname
+        // drop the guaranteed leading slash, so that we don't produce an empty segment for it
```

Comment:
Hm, I think the "guarantee" is not strong enough for me - it's not a `NonEmptyList` after all.

I think it's better to do `url.pathname.split("/").dropWhile(_.isEmpty)` - it should be safer.

Reply:
I took the guarantee from the Scaladoc, https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname also repeats this claim... but yeah I can still add that

Reply:
actually, dropWhile will drop empty segments if they precede non-empty ones, I think that'll break a test or two. Having a guaranteed slash before each segment is one way to ensure we don't accidentally drop segments. The spec seems pretty unambiguous on that part:

> For each segment of url’s path: append U+002F (/) followed by segment to output

Reply:
right, I sort of hoped empty segments were disallowed in general 🙃